lib/AST/Decl.cpp

```diff
@@ -1132,6 +1132,69 @@ VarDecl *PatternBindingDecl::getSingleVar() const {
   return nullptr;
 }
 
+/// Check whether the given type representation will be
+/// default-initializable.
+static bool isDefaultInitializable(TypeRepr *typeRepr) {
```

Comment:
We said we didn't really want to expose this on `TypeRepr` directly, so this is copied as-is to here.